fix(board-core): Set background-color to empty string in default styles for canvas and window

0xRezaa opened the pull request:
Background color passed via the environmentProps is still applied for canvas and window.

PeterShershov commented:
@0xRezaa 
Notice that `applyStylesToWindow` still adds this value to the body, so if this value gets passed via a class it should actually be a variable. We can just remove it if it shouldn't be applied by default.

0xRezaa commented:
> @0xRezaa Notice that `applyStylesToWindow` still adds this value to the body, so if this value gets passed via a class it should actually be a variable. We can just remove it if it shouldn't be applied by default.

When I replace the backgroundColor in the `defaultWindowStyles` with an empty string there won't be any value set as inline style by `applyStylesToWindow`. So replacing the value with an empty string works for both window and canvas default styles.

But I am not sure what you mean with the value being a variable. Can you elaborate on that?
As far as I understand the user has the option to specify cavas and window related styles in the `environmentProps` of a simulation/board. My suggestion was putting the default background colors in a class-selector which would still allow the user to pass the background color via the `environmentProps` which would be applied as inline styles and thus overwrite the defaults.

Do you think it would be appropriate to add a style tag string in `setupSimulationStage` like so:
```
const SIMULATION_CANVAS_ID = "simulation-canvas"
const style = `<style>
    body {
        background-color: #fcfcfc;
    }

    .${SIMULATION_CANVAS_ID} {
        background-color: #fff;
     }
    </style>`
document.head.insertAdjacentHTML('beforeend', style);
```

or instead to move the default background colors into the simulation-preview feature where a [style tag like above](https://github.com/wixplosives/component-studio/blob/6f690f8e95043e1997de4e55c26944784b4131e5/packages/simulation-preview/src/preview/simulation-window-controller.ts#L95-L123) is already added

0xRezaa commented:
@PeterShershov 
I have pushed a version of what I have suggested

AviVahl commented:
Can we completely remove forcing the background color?
In addition, I'd suggest to never use an id and an id selector for a UI piece that could _potentially_ be rendered multiple times in a page (by direct consumers of the board API).

0xRezaa commented:
> Can we completely remove forcing the background color? In addition, I'd suggest to never use an id and an id selector for a UI piece that could _potentially_ be rendered multiple times in a page (by direct consumers of the board API).

I removed forcing the background color as inline style.
Removing the background color completely (also as a style element with class selector) we lose the background color of the canvas (see attached picture) so I am not sure if that is what you meant by "remove forcing the background color".

I changed the selector from a id to class and also created a class name constant which can be imported and replace the one in ./component-studio/packages/simulation-preview/src/universal/constants.ts

talh78 commented:
Closes https://github.com/wixplosives/component-studio/issues/9389

0xRezaa commented:
After getting rid (see [here](https://github.com/wixplosives/wcs-core/pull/493/files#diff-a7930e8dee274a83b04142d7e005a78eb782c3d9ff9cb8ba7b9928ca23288c8fL37)) of setting the `windowBackgroundColor` in the `previousWindowEnvironmentProps` should we also get rid of these [lines](https://github.com/wixplosives/wcs-core/blob/a64ea5d5194780b8a91ffd23713157433bb7fc6f/packages/board-core/src/setup-stage.tsx#L129-L131) in the cleanup?

PeterShershov commented on packages/board-core/src/setup-stage.tsx:
```diff
@@ -93,7 +89,16 @@ const applyStylesToCanvas = (canvas: HTMLDivElement, environmentProps: ICanvasEn
-    canvas.setAttribute('id', 'board-canvas');
+    const style = `<style>
+        body {
+            background-color: #fcfcfc;
```
let's put these guys into variables

PeterShershov commented:
@0xRezaa I'm not even sure why we needed this functionality.

0xRezaa commented:
Closed in favour of https://github.com/wixplosives/component-studio/pull/10009 and https://github.com/wixplosives/wcs-core/pull/580
